src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.PublishVerticalManifest/PublishVerticalManifest.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.SourceBuild.Tasks.PublishVerticalManifest
+{
+    public class PublishVerticalManifest : Task
+    {
+        public string AssetManifestsLocalStorageDir { get; set; }
+        public string VerticalAssetManifestLocalPath { get; set; }
+
+        public override bool Execute()
+        {
+            var assetManifestXmls = Directory.GetFiles(AssetManifestsLocalStorageDir, "*.xml", SearchOption.AllDirectories)
```

Comment:
I would rather see these passed in as an `ItemGroup` rather than a pattern search. It's more controllable.